src/app/app-component.service.ts

```diff
@@ -0,0 +1,189 @@
+import { Injectable } from '@angular/core';
+import { weatherData, widgetListInterface, widgetWeatherData } from './interfaces';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AppComponentService {
+
+  constructor() { }
+
+  weatherDataTransformation(inputData: weatherData): widgetWeatherData[] {
+
+    let limitDays = 5; //НАСТРОЙКА КОЛЛИЧЕСТВА ДНЕЙ ДЛЯ ПОКАЗА
+    let dayCounter: number = -1; //СЧЕТЧИК ДНЕЙ ДЛЯ ПОКАЗА
+    let currentDate = new Date(0); //ТЕКУЩАЯ ДАТА ДЛЯ РАЗДЕЛЕНИЯ ПО ДНЯМ
+
+    let outputData: widgetWeatherData[] = new Array();
+
+
+
+    for (let i = 0; i < inputData.list.length; i++) { //ПЕРЕБОР ВСЕХ ВРЕМЕННЫХ МЕТОК
+
+      if ((new Date(inputData.list[i].dt_txt)).getDate() != currentDate.getDate()) {
+
+        if (dayCounter != -1) {
+          this.averagingTemperature(outputData[dayCounter]);
+        }; //УСРЕДНЕНИЕ ТЕМПЕРАТУРЫ ЗА ДЕНЬ, ПЕРЕД ПЕРЕХОДОМ К СЛЕДУЮЩЕМУ
+
+        dayCounter++;
+
+        if (dayCounter === limitDays) break; //ПРОВЕРКА ЛИМИТА ПОКАЗЫВАЕМЫХ ДНЕЙ
+
+        outputData.push({
+          day: "",
+          number: 0,
+          month: "",
+          temp_max: 0,
+          temp_min: 0,
+          list: []
+        });
+
+
+        currentDate = new Date(inputData.list[i].dt_txt);
+
+        outputData[dayCounter].day = this.matchingDay((new Date(inputData.list[i].dt_txt)).getDay());
+        outputData[dayCounter].number = (new Date(inputData.list[i].dt_txt)).getDate();
+        outputData[dayCounter].month = this.matchingMonth((new Date(inputData.list[i].dt_txt)).getMonth());;
+
+      }; //СМЕНА ДНЯ НА СЛЕДУЮЩИЙ
+
+
+
+      outputData[dayCounter].list.push(this.objectTransformation(inputData.list[i]));
+      //КЛАДЕМ В ТЕКУЩИЙ ДЕНЬ ИЗМЕНЕННУЮ ВРЕМЕННУЮ МЕТКУ
+    };
+
+    return outputData;
+  };
+
+  averagingTemperature(inputObject: widgetWeatherData) {
+    let minTemperature: number = 0;
+    let maxTemperature: number = 0;
+    inputObject.list.forEach(element => {
+      if (element.temp > maxTemperature) maxTemperature = element.temp;
+      if (element.temp < minTemperature) minTemperature = element.temp;
+    });
+    inputObject.temp_min = Math.floor(minTemperature);
+    inputObject.temp_max = Math.floor(maxTemperature);
+  };
+
+  matchingDay(inputData: number): string {
```

Comment:
What about simple object?
1. Create enum where you will declare all weekdays
e.g.
```ts
enum Weekday {
    Monday = 0,
    Tuesday,
    Wednesday
...
```

2. create an object with your 'localization' of those days.
```
const RUSSIAN_WEEKDAY_LOCALE: Record<Weekday, string> = {
    [Weekday.Monday]: 'Понедельник',
    [Weekday.Tuesday]: 'Вторник',
...
```